public/app/features/alerting/unified/components/receivers/form/fields/TemplateSelector.test.tsx

```diff
@@ -0,0 +1,35 @@
+import { getTemplateOptions } from './TemplateSelector';
+
+describe('getTemplateOptions function', () => {
+  it('should return the last template when there are duplicates', () => {
+    const templateFiles = {
+      file1:
+        '{{ define "template1" }}{{ len .Alerts.Firing }} firing alert(s), {{ len .Alerts.Resolved }} resolved alert(s){{ end }}',
+      // duplicated define, the last one should be returned
+      file2:
+        '{{ define "template1" }}{{ len .Alerts.Firing }} firing alert(s), {{ len .Alerts.Resolved }} resolved alert(s) this is the last one{{ end }}',
+      file3:
+        '{{ define "email.subject" }}{{ len .Alerts.Firing }} firing alert(s), {{ len .Alerts.Resolved }} resolved alert(s){{ end }}',
+      // define with a minus sign
+      file4: '{{ define "template_with_minus" -}}{{ .Annotations.summary }}{{- end }}',
+      file5: '',
+      file6: `{{ define "nested" }}
+      Main Template Content
+      {{ template "sub1" }}
+      {{ template "sub2" }}
+      {{ end }}
+      
+      {{ define "sub1" }}
+      Sub Template 1 Content
+      {{ end }}
+      
+      {{ define "sub2" }}
+      Sub Template 2 Content
+      {{ end }}`,
+    };
+
+    const result = getTemplateOptions(templateFiles);
+
+    expect(result).toMatchSnapshot();
```

Comment:
Is there a nicer assertion we can make for this? As the snapshot is a little unclear, I wouldn't know what to expect if it changed?

Reply:
I was not using snapshot at the beginning of this pr, but given the text of the result is soo long, i think it's a good option to use the snapshot. This snapshot should not change. If it changes, something is wrong.

Reply:
I was thinking something more like this, so it's clearer to see what the test is checking:

```ts
const result = getTemplateOptions(templateFiles);

const template1Matches = result.filter((option) => option.label === 'template1');
expect(template1Matches).toHaveLength(1);
expect(template1Matches[0].value?.content).toMatch(/this is the last one/i);

const file5Matches = result.filter((option) => option.label === 'file5');
expect(file5Matches).toHaveLength(0);
```

What do you think?

Reply:
I still think in this case it's better to use snapshot as we want the result to be exactly like the snapshot is. I aded description in the templateFiles const with the different use cases we are referencing so, we know what we are testing in each template file.